src/instance.ts

```diff
@@ -415,24 +552,39 @@ class Instance extends common.ServiceObject {
    *   const apiResponse = data[0];
    * });
    */
-  get(options, callback?) {
-    if (is.fn(options)) {
-      callback = options;
-      options = {};
-    }
+  get(
+    optionsOrCallback?: GetConfig | GetInstanceCallback,
+    cb?: GetInstanceCallback
+  ): void | Promise<GetInstanceResponse> {
+    const callback =
+      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb!;
+    const options =
+      typeof optionsOrCallback === 'object'
+        ? optionsOrCallback
+        : ({} as GetConfig);
+
     this.getMetadata((err, metadata) => {
       if (err) {
         if (err.code === 5 && options.autoCreate) {
-          this.create(options, (err, database, operation) => {
-            if (err) {
-              callback(err);
-              return;
+          this.create(
+            options,
+            (
+              err: ServiceError | null,
```

Comment:
I think we should be able to infer these types?

Reply:
here no definition for `this.create`, so need to expose datatype..